FF1Lib/Scale.cs

```diff
@@ -48,17 +48,25 @@ public void ScalePrices(IScaleFlags flags, Blob[] text, MT19337 rng)
 				var newPrice = Scale(prices[i] / multiplier, scale, 1, rng);
 				prices[i] = (ushort) (flags.WrapPriceOverflow ? ((newPrice - 1) % 0xFFFF) + 1 : Min(newPrice, 0xFFFF));
             }
-            var questItemPrice = prices[(int)Item.Bottle];
-            for (var i = 0; i < (int)Item.Tent; i++)
-            {
-                prices[i] = questItemPrice;
-            }
+			var questItemPrice = prices[(int)Item.Bottle];
+			// If we don't do this before checking for the item shop location factor, Ribbons and Shirts will end up being really cheap
+			// This realistically doesn't matter without Shop Wares shuffle on because nobody wants to sell Ribbons/Shirts, but if it is...
 			prices[(int)Item.WhiteShirt] = (ushort)(questItemPrice / 2);
 			prices[(int)Item.BlackShirt] = (ushort)(questItemPrice / 2);
 			prices[(int)Item.Ribbon] = questItemPrice;
-            // Crystal can block Ship in early game where 50000 G would be too expensive
-            prices[(int)Item.Crystal] = (ushort)(prices[(int)Item.Crystal] / 8);
-
+			var itemShopFactor = new Dictionary<MapLocation, int>() {
+				{ MapLocation.Coneria, 8 },
+				{ MapLocation.Pravoka, 2 }
+			};
+			var divisor = new int();

```

Comment:
I still don't think you need this line - can you do out int divisor in the following TryGetValue?